data-model-translator-wrapper/src/main/java/ai/aitia/arrowhead/dmtpw/service/DataModelTranslationService.java

```diff
@@ -0,0 +1,162 @@
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.dto.DataModelTranslationInitRequestDTO;
+import eu.arrowhead.dto.DataModelTranslationResultResponseDTO;
+import eu.arrowhead.dto.enums.DataModelTranslationTaskStatus;
+import jakarta.annotation.PostConstruct;
+
+@Service
+public class DataModelTranslationService {
+
+	//=================================================================================================
+	// members
+	
+	@Autowired
+	private DataModelTranslatorWrapperSystemInfo sysInfo;
+
+	@Autowired
+	private DataModelTranslationServiceValidation validator;
+
+	private final BlockingQueue<DataModelTranslationTask> jobQueue = new LinkedBlockingQueue<>();
+	private final Map<UUID, DataModelTranslationTask> jobCache = new ConcurrentHashMap<>();
+	private final Map<UUID, String> fileNameCache = new ConcurrentHashMap<>();
+
+	private final Logger logger = LogManager.getLogger(this.getClass());
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+    public UUID initTranslation(final DataModelTranslationInitRequestDTO dto, final String origin) {
+    	logger.debug("initTranslationJob started...");
+    	
+    	final DataModelTranslationInitRequestDTO normalized = validator.validateAndNormalizeDataModelTranslationInitRequestDTO(dto, origin);
+
+        final DataModelTranslationTask job = new DataModelTranslationTask(sysInfo.getModelIdsWithResultMimeTpyes().get(List.of(normalized.inputModelId(), normalized.outputModelId())), normalized.payload());
+        jobQueue.add(job);
+        jobCache.put(job.getUuid(), job);
+        return job.getUuid();
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    public DataModelTranslationResultResponseDTO getTranslationResult(final UUID jobId, final String origin) {
+    	logger.debug("getTranslationResult started...");
+
+        final DataModelTranslationTask job = jobCache.get(jobId);
+        
+        if (job == null) {
+        	throw new InvalidParameterException("Invalid UUID", origin);
+        }
+
+        final String result = readTranslatedFile(fileNameCache.get(jobId));
+        return new DataModelTranslationResultResponseDTO(job.getStatus(), result, job.getResultMimeType());
+    }
+
+	//=================================================================================================
+	// assistant methods
+
+    //-------------------------------------------------------------------------------------------------
+    private void doTranslationJob(final DataModelTranslationTask job) {
+    	// start the job
+    	job.setStatus(DataModelTranslationTaskStatus.IN_PROGRESS);
+
+    	// decode the bytes
+    	final byte[] bytes = Base64.getDecoder().decode(job.getPayload());
```

Comment:
you use wrong character set here. Use something similar than this one:

`final byte[] bytes = Base64.getDecoder().decode(job.getPayload().getBytes(StandardCharsets.UTF_8));`